examples/a2c/a2c.py

```diff
@@ -2,164 +2,144 @@
+    record_interval = cfg.logger.log_interval
+    pbar = tqdm.tqdm(total=cfg.collector.total_frames)
+    collected_frames = 0
+
+    # Main loop
+    r0 = None
+    l0 = None
+    for data in collector:
+
+        frames_in_batch = data.numel()
+        collected_frames += frames_in_batch * cfg.env.frame_skip
+        pbar.update(data.numel())
+        data_view = data.reshape(-1)
+
+        # Compute GAE
+        with torch.no_grad():
+            batch = adv_module(data_view)
+
+        # Normalize advantage
+        adv = batch.get("advantage")
+        loc = adv.mean().item()
+        scale = adv.std().clamp_min(1e-6).item()
+        adv = (adv - loc) / scale
+        batch.set("advantage", adv)
+
+        # Forward pass A2C loss
+        batch = batch.to(model_device)
+        loss = loss_module(batch)
+        loss_sum = loss["loss_critic"] + loss["loss_objective"] + loss["loss_entropy"]
+
+        # Backward pass + learning step
+        loss_sum.backward()
+        grad_norm = torch.nn.utils.clip_grad_norm_(
+            list(actor.parameters()) + list(critic.parameters()), max_norm=0.5
+        )
+        optim.step()
+        if scheduler is not None:
+            scheduler.step()
+        optim.zero_grad()
+
+        # Logging
+        if r0 is None:
+            r0 = data["next"]["reward"].mean().item()
+        if l0 is None:
+            l0 = loss_sum.item()
+        pbar.set_description(
+            f"loss: {loss_sum.item(): 4.4f} (init: {l0: 4.4f}), reward: {data['next']['reward'].mean(): 4.4f} (init={r0: 4.4f})"
+        )
+        if logger is not None:
+            for key, value in loss.items():
+                logger.log_scalar(key, value.item(), collected_frames)
+            logger.log_scalar("grad_norm", grad_norm.item(), collected_frames)
+            episode_rewards = data["next"]["episode_reward"][data["next"]["done"]]
+            if len(episode_rewards) > 0:
+                logger.log_scalar(
+                    "reward_training", episode_rewards.mean().item(), collected_frames
+                )
+        collector.update_policy_weights_()
+
+        # Test current policy
+        if (
+            logger is not None
+            and (collected_frames - frames_in_batch) // record_interval
+            < collected_frames // record_interval
+        ):
+
+            with torch.no_grad():
+                test_env.eval()
+                actor.eval()
+                # Generate a complete episode
+                td_test = test_env.rollout(
+                    policy=actor,
+                    max_steps=10_000_000,
+                    auto_reset=True,
+                    auto_cast_to_device=True,
+                    break_when_any_done=True,
+                ).clone()
+                logger.log_scalar(
+                    "reward_testing",
+                    td_test["next"]["reward"].sum().item(),
+                    collected_frames,
+                )
+                actor.train()
```

Comment:
Nit: If we use the ``Recorder`` class we can do all of this I guess, but it's a bit of a black box so I'm ok with the explicit calls.
Maybe let's we use `td_test["next", "reward"]` when we can (same for all the key indexing in the script) :)

Reply:
I saw that in Recorder you need to specify a number of steps, and I wanted to record a single test episode, independently of the number of steps. Maybe recorder could accept a number of episodes to records instead of a number steps?